interfaces/builtin/mount_control.go

```diff
@@ -0,0 +1,320 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/utils"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/strutil"
+	"github.com/snapcore/snapd/systemd"
+)
+
+const mountControlSummary = `allows creating transient and persistent mounts`
+
+const mountControlBaseDeclarationSlots = `
+  mount-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+        - gadget
+    deny-connection: true
+`
+
+const mountAttrNotADictionary = `mount-control "mount" attribute is not a dictionary`
+const mountControlConnectedPlugSecComp = `
+# Description: Allow mount and umount syscall access.
+mount
+umount
+umount2
+`
+
+var allowedMountOptions = []string{
+	"async",
+	"atime",
+	"bind",
+	"diratime",
+	"dirsync",
+	"iversion",
+	"lazytime",
+	"mand",
```

Comment:
I don't think we should allow `mand` - according to the `fcntl(2)` man page, the Linux implementation of mandatory locking is buggy so I fear this could lead to a state where we are unable to unmount a file-system. So I think we should not allow this unless there is a strong use-case for it.

Reply:
Removed.